DEB package does not remove binaries/configuration

## General Information

- This is a: bug report
- Captain version: `main`
- OS: Debian 10 x64 LXC
- Proxmox version: `v6.3-3`

## Issue Report

```
2:05PM DBG building new plane PlaneName=tfwip1.firecore.lab
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xae4d02]

goroutine 11 [running]:
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetNextID(0x0, 0x0, 0x2b, 0xc00029f250, 0xb)
        /root/go/pkg/mod/github.com/!telmate/proxmox-api-go@v0.0.0-20210517153043-5b9c621ea0cd/proxmox/client.go:766 +0x72
github.com/ARMmaster17/Captain/ATC/drivers/providers.ProxmoxLxcProviderDriver.BuildPlane(0x0, 0xc0003944b0, 0x0, 0x0, 0xc0003944b0, 0xc0003b97a8)
        /src/Captain/ATC/drivers/providers/ProxmoxLxcProviderDriver.go:71 +0x7de
github.com/ARMmaster17/Captain/ATC/drivers.BuildPlaneOnAnyProvider(0xc0003944b0, 0xc0003944b0, 0x0, 0x0, 0x13)
        /src/Captain/ATC/drivers/ProviderDriverManager.go:24 +0x21b
main.(*Plane).buildPlane(0xc0002db900, 0x0, 0x0)
        /src/Captain/ATC/Plane.go:108 +0x1d4
main.(*Plane).BeforeCreate(0xc0002db900, 0xc00039ddd0, 0xc0002db900, 0x7f7f01447c80)
        /src/Captain/ATC/Plane.go:78 +0x2da
gorm.io/gorm/callbacks.BeforeCreate.func1(0xcc0d40, 0xc0002db900, 0xc00039ddd0, 0xcc0d01)
        /root/go/pkg/mod/gorm.io/gorm@v1.21.10/callbacks/create.go:25 +0xc2
gorm.io/gorm/callbacks.callMethod(0xc00039cc60, 0xc0003b9a08)
        /root/go/pkg/mod/gorm.io/gorm@v1.21.10/callbacks/callmethod.go:20 +0x2d1
gorm.io/gorm/callbacks.BeforeCreate(0xc00039cc60)
        /root/go/pkg/mod/gorm.io/gorm@v1.21.10/callbacks/create.go:14 +0x8e
gorm.io/gorm.(*processor).Execute(0xc00033d9a0, 0xc00039cc60, 0xd24b4a)
        /root/go/pkg/mod/gorm.io/gorm@v1.21.10/callbacks.go:130 +0x447
gorm.io/gorm.(*DB).Save(0xc000379860, 0xcc0d40, 0xc0002db900, 0x9)
        /root/go/pkg/mod/gorm.io/gorm@v1.21.10/finisher_api.go:87 +0x72f
main.builder.buildPlane(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /src/Captain/ATC/Builder.go:44 +0x1f7
created by main.(*Formation).launchBuilder
        /src/Captain/ATC/Formation.go:121 +0x257
```

Just a guess, this is because the base objects for each method are not pointer objects, so the session object is initialized, but then lost when the method returns.

--- a/ATC/drivers/ProviderDriverManager.go
+++ b/ATC/drivers/ProviderDriverManager.go
@@ -75,7 +75,7 @@ func getDestroyDriver(cuid string) (providers.ProviderDriver, error) {
 func driverLookupByYAMLTag(driverentry string) (providers.ProviderDriver, error) {
 	if driverentry == (&providers.ProxmoxLxcProviderDriver{}).GetYAMLTag() {
 		// In the future there could be a check here to see if a driver is ready to receive new planes.
-		return providers.ProxmoxLxcProviderDriver{}, nil
+		return &providers.ProxmoxLxcProviderDriver{}, nil
 	}
 	if driverentry == (&providers.DummyProviderDriver{}).GetYAMLTag() {
 		return providers.DummyProviderDriver{}, nil
@@ -87,7 +87,7 @@ func driverLookupByYAMLTag(driverentry string) (providers.ProviderDriver, error)
 func driverLookupByCUIDPrefix(cuidprefix string) (providers.ProviderDriver, error) {
 	if cuidprefix == (&providers.ProxmoxLxcProviderDriver{}).GetCUIDPrefix() {
 		// In the future there could be a check here to see if a driver is ready to receive new planes.
-		return providers.ProxmoxLxcProviderDriver{}, nil
+		return &providers.ProxmoxLxcProviderDriver{}, nil
 	}
 	if cuidprefix == (&providers.DummyProviderDriver{}).GetCUIDPrefix() {
 		return providers.DummyProviderDriver{}, nil

--- a/ATC/drivers/providers/ProxmoxLxcProviderDriver.go
+++ b/ATC/drivers/providers/ProxmoxLxcProviderDriver.go
@@ -18,7 +18,7 @@ type ProxmoxLxcProviderDriver struct {
 
 // Connect reads any Proxmox connection parameters defined in the environment and performs authentication with the
 // designated Proxmox cluster.
-func (d ProxmoxLxcProviderDriver) Connect() error {
+func (d *ProxmoxLxcProviderDriver) Connect() error {
 	tlsConf := &tls.Config{InsecureSkipVerify: !viper.GetBool(d.getConfigItemPath("forcessl"))}
 	d.client, _ = proxmox.NewClient(viper.GetString(d.getConfigItemPath("url")), nil, tlsConf, 300)
 	return d.client.Login(os.Getenv("CAPTAIN_PROXMOX_USER"), os.Getenv("CAPTAIN_PROXMOX_PASSWORD"), "")
@@ -26,7 +26,7 @@ func (d ProxmoxLxcProviderDriver) Connect() error {
 
 // BuildPlane converts the given GenericPlane into a data structure that is usable by the Proxmox API and submits
 // the container for provisioning.
-func (d ProxmoxLxcProviderDriver) BuildPlane(p *GenericPlane) (string, error) {
+func (d *ProxmoxLxcProviderDriver) BuildPlane(p *GenericPlane) (string, error) {
 	config := proxmox.NewConfigLxc()
 	template, err := imagestore.GetProviderSpecificImageConfiguration(d.GetYAMLTag(), viper.GetString("defaults.image"))
 	if err != nil {
@@ -78,7 +78,7 @@ func (d ProxmoxLxcProviderDriver) BuildPlane(p *GenericPlane) (string, error) {
 }
 
 // DestroyPlane will destroy a plane that is managed by the Proxmox LXC driver.
-func (d ProxmoxLxcProviderDriver) DestroyPlane(cuid string, p *GenericPlane) error {
+func (d *ProxmoxLxcProviderDriver) DestroyPlane(cuid string, p *GenericPlane) error {
 	vmr, err := d.client.GetVmRefByName(p.FQDN)
 	if err != nil {
 		return fmt.Errorf("unable to obtain reference to underlying LXC container for plane %s:\n%w", p.FQDN, err)
@@ -96,24 +96,24 @@ func (d ProxmoxLxcProviderDriver) DestroyPlane(cuid string, p *GenericPlane) err
 
 // GetCUIDPrefix gets the prefix that should be added to the beginning of the CUID strings for all planes that are
 // managed by this driver.
-func (d ProxmoxLxcProviderDriver) GetCUIDPrefix() string {
+func (d *ProxmoxLxcProviderDriver) GetCUIDPrefix() string {
 	return "proxmox.lxc"
 }
 
 // GetYAMLTag gets the YAML tag that the Proxmox LXC driver uses to identify settings unique to this driver
 // in config.yaml.
-func (d ProxmoxLxcProviderDriver) GetYAMLTag() string {
+func (d *ProxmoxLxcProviderDriver) GetYAMLTag() string {
 	return "proxmoxlxc"
 }
 
 // getFullYAMLTag gets the base YAML tag for all settings that are unique to the Proxmox LXC driver.
-func (d ProxmoxLxcProviderDriver) getFullYAMLTag() string {
+func (d *ProxmoxLxcProviderDriver) getFullYAMLTag() string {
 	return fmt.Sprintf("drivers.provisioners.%s", d.GetYAMLTag())
 }
 
 // getConfigItemPath returns the full YAML path to a configuration file for the Proxmox LXC driver with the given tag(s) appended
 // at the end.
-func (d ProxmoxLxcProviderDriver) getConfigItemPath(entryPath string) string {
+func (d *ProxmoxLxcProviderDriver) getConfigItemPath(entryPath string) string {
 	return fmt.Sprintf("%s.%s", d.getFullYAMLTag(), entryPath)
 }
 

